FSI/FSI.C

```diff
@@ -72,88 +72,29 @@ std::string preciceAdapter::FSI::FluidStructureInteraction::determineSolverType(
     {
-        transportPropertiesExists = true;
-        DEBUG(adapterInfo("Found the transportProperties dictionary."));
-    }
-    else
-    {
-        DEBUG(adapterInfo("Did not find the transportProperties dictionary."));
-    }
+      volScalarField p_ = mesh_.lookupObject<volScalarField>("p");
 
-    
-    if (mesh_.foundObject<IOdictionary>(turbulenceModel::propertiesName))
-    {
-        turbulencePropertiesExists = true;
-        DEBUG(adapterInfo("Found the " + turbulenceModel::propertiesName
-            + " dictionary."));
+      if (p_.dimensions() == pressureDimensionsCompressible)
+        solverType = "compressible";
+      else if (p_.dimensions() == pressureDimensionsIncompressible)
+        solverType = "incompressible";
     }
-    else
-    {
-        DEBUG(adapterInfo("Did not find the " + turbulenceModel::propertiesName
-            + " dictionary."));
-    }
-    
 
-    if (mesh_.foundObject<IOdictionary>("thermophysicalProperties"))
-    {
-        thermophysicalPropertiesExists = true;
-        DEBUG(adapterInfo("Found the thermophysicalProperties dictionary."));
-    }
-    else
-    {
-        DEBUG(adapterInfo("Did not find the thermophysicalProperties dictionary."));
-    }
-
-    if (turbulencePropertiesExists)
-    {
-        if (thermophysicalPropertiesExists)
-        {
-            solverType = "compressible";
-            DEBUG(adapterInfo("This is a compressible flow solver, "
-                "as turbulence and thermophysical properties are provided."));
-        }
-        else if (transportPropertiesExists)
-        {
-            solverType = "incompressible";
-            DEBUG(adapterInfo("This is an incompressible flow solver, "
-            "as turbulence and transport properties are provided."));
-        }
-        else
-        {
-            adapterInfo("Could not determine the solver type, or this is not "
-            "a compatible solver: although turbulence properties are provided, "
-            "neither transport or thermophysical properties are provided.",
-            "error");
-        }
-    }
-    else
-    {
-        if (transportPropertiesExists)
-        {
-            solverType = "basic";
-            DEBUG(adapterInfo("This is a basic solver, as transport properties "
-            "are provided, while turbulence or transport properties are not "
-            "provided."));
-        }
-        else
-        {
-            adapterInfo("Could not determine the solver type, or this is not a "
-            "compatible solver: neither transport, nor turbulence properties "
-            "are provided.",
-            "error");
-        }
-    }
+    if (solverType == "unknown")
+      adapterInfo("Failed to determine the solver type. "
+                  "Please specify your solver type in the preciceDict. "
+                  "Known solver types are: "
+                  "Incompressible"
+                  "Compressible"
+                  "Basic",
+                  "error");
```

Comment:
How are we treating the `error` solver type? Is this something that we want to show to the user?

Reply:
Hm this was my understanding of your proposal in the discussion above:
> If none of these can be determined (i.e. there is no pressure field), we can either:
say "unknown solver type, please define it in preciceDict"

Reply:
I just realized that `error` is just the usual argument to `adapterInfo()`. I got confused by the different lines.

Reply:
Looks good then!

Reply:
Alright, then I will test everything and merge, if it is fine.